### PR TITLE
Bump Dart SDK to 1.22.0-dev.10.3

### PR DESCRIPTION
**Pull request by goderbauer**
Required for engine roll (https://github.com/flutter/flutter/pull/7777).

### CODE REVIEW

File: examples/stocks/lib/stock_data.dart [hunk 1]
```diff
@@ -76,7 +76,7 @@ class StockDataFetcher {
       String json = response.body;
       if (json == null) {
         print("Failed to load stock data chunk ${_nextChunk - 1}");
-        return;
+        return null;
```
Comment (Hixie) [inline examples/stocks/lib/stock_data.dart]: This shouldn't be needed. `return` without a value should just return null, no? We rely on this all over the place.

File: packages/flutter_test/lib/src/controller.dart [hunk 2]
```diff
@@ -143,7 +143,7 @@ class WidgetController {
   ///
   /// * Use [firstState] if you expect to match several states but only want the first.
   /// * Use [stateList] if you expect to match several states and want all of them.
-  T state<T extends State>(Finder finder) {
+  T state<T extends State<StatefulWidget>>(Finder finder) { // TODO(leafp) remove '<StatefulWidget>' when https://github.com/dart-lang/sdk/issues/28580 is fixed
```
Comment (Hixie) [inline packages/flutter_test/lib/src/controller.dart]: nit: the syntax is `// TODO(username): message url`, you're missing the colon.

Comment (Hixie) [inline packages/flutter_test/lib/src/controller.dart]: x4

File: packages/flutter_test/lib/src/test_async_utils.dart [hunk 3]
```diff
@@ -102,8 +102,10 @@ class TestAsyncUtils {
         throw new FlutterError(message.toString().trimRight());
       }
     }
-    return result.then(
-      (Null value) => completionHandler(null, null),
+    return result.then<Null>(
+        (Null value) {
```
Comment (Hixie) [inline packages/flutter_test/lib/src/test_async_utils.dart]: this is over-indented by 2 (should line up with the onError)

**Comment by pq** [comment]
Once you get it past @Hixie, works for me. 👍  Thanks for picking this up!

File: examples/stocks/lib/stock_data.dart (hunk 1 above)
Comment (goderbauer) [inline examples/stocks/lib/stock_data.dart]: The analyzer is complaining about it:
```
[warning] Missing return value after 'return'. (/usr/local/google/home/goderbauer/dev/flutter/examples/stocks/lib/stock_data.dart, line 79, col 9)
```

Comment (Hixie) [inline examples/stocks/lib/stock_data.dart]: we talked about it and i am convinced we can work with this change

**Comment by Hixie** [comment]
LGTM

**Pull request merged**